### PR TITLE
pr-publish: use workflow_dispatch trigger

### PR DESCRIPTION
**Pull request by dawidd6**
- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This will create an option to trigger bottle publish via GitHub UI, as shown here: https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/

Needs to be merged in sync with: https://github.com/Homebrew/homebrew-core/pull/58725

### CODE REVIEW

File: Library/Homebrew/dev-cmd/pr-publish.rb [hunk 1]
```diff
@@ -16,6 +16,10 @@ def pr_publish_args
       EOS
       flag   "--tap=",
              description: "Target tap repository (default: `homebrew/core`)."
+      flag   "--workflow=",
+             description: "Target workflow filename (default: `publish-commit-bottles.yml`)."
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/pr-publish.rb:20]: When would you use a different name here?

File: Library/Homebrew/dev-cmd/pr-publish.rb [hunk 2]
```diff
@@ -16,6 +16,10 @@ def pr_publish_args
...
+      flag   "--ref=",
+             description: "Target ref name (default: `master`)."
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/pr-publish.rb:22]: When would you use a different branch here?

File: Library/Homebrew/dev-cmd/pr-publish.rb (hunk 1 above)
Comment (dawidd6) [reply Library/Homebrew/dev-cmd/pr-publish.rb:20]: Probably for other third-party taps, this option might be useful. I'm going to actually use it in mine.

File: Library/Homebrew/dev-cmd/pr-publish.rb (hunk 2 above)
Comment (dawidd6) [reply Library/Homebrew/dev-cmd/pr-publish.rb:22]: Good question. For now I don't see any usages. Can just delete this option altogether and default to `master` always.

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/pr-publish.rb:22]: Yeh, just delete for now I reckon then 👍🏻

File: Library/Homebrew/dev-cmd/pr-publish.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/pr-publish.rb:20]: Cool. I wonder if it can be inferred somehow 🤔. Don't let that block this PR.

Comment (dawidd6) [reply Library/Homebrew/dev-cmd/pr-publish.rb:20]: Hmm, nothing that I am aware of as of now, but I get the idea.

**Pull request merged**